pyuvdata/utils.py

```diff
@@ -1518,27 +1529,258 @@ def uvcalibrate(
+            uvcal_times_to_keep = []
+            for this_time in uvdata_times:
+                wh_time_match = np.nonzero(
+                    np.isclose(
+                        uvcal.time_array - this_time,
+                        0,
+                        atol=uvdata._time_array.tols[1],
+                        rtol=uvdata._time_array.tols[0],
+                    )
+                )
+                if wh_time_match[0].size > 0:
+                    uvcal_times_to_keep.append(uvcal.time_array[wh_time_match][0])
+                else:
+                    warnings.warn(
+                        f"Time {this_time} exists on UVData but not on UVCal. "
+                        "This will become an error in version 2.2",
+                        DeprecationWarning,
+                    )
+            if len(uvcal_times_to_keep) < uvcal.Ntimes:
+                downselect_cal_times = True
+    elif uvcal.time_range is None:
+        if uvdata.Ntimes > 1 or not np.isclose(
+            uvdata_times,
+            uvcal.time_array,
+            atol=uvdata._time_array.tols[1],
+            rtol=uvdata._time_array.tols[0],
+        ):
+            if override_time_check:
+                warnings.warn(
+                    "Times do not match between UVData and UVCal "
+                    "but override_time_check is True, so calibration "
+                    "will be applied anyway."
+                )
+            else:
+                warnings.warn(
+                    "Times do not match between UVData and UVCal. "
+                    "Set the override_time_check keyword to apply calibration anyway. "
+                    "This will become an error in version 2.2",
+                    DeprecationWarning,
+                )
+    else:
+        # time_array is length 1 and time_range exists: check uvdata_times in time_range
+        if (
+            np.min(uvdata_times) < uvcal.time_range[0]
+            or np.max(uvdata_times) > uvcal.time_range[1]
+        ):
+            if override_time_check:
+                warnings.warn(
+                    "Times do not match between UVData and UVCal "
+                    "but override_time_check is True, so calibration "
+                    "will be applied anyway."
+                )
+            else:
+                warnings.warn(
+                    "Times do not match between UVData and UVCal. "
+                    "Set the override_time_check keyword to apply calibration anyway. "
+                    "This will become an error in version 2.2",
+                    DeprecationWarning,
+                )
+
+    downselect_cal_freq = False
+    try:
+        freq_arr_match = np.allclose(
+            np.sort(uvcal.freq_array[0, :]),
+            np.sort(uvdata.freq_array[0, :]),
+            atol=uvdata._freq_array.tols[1],
+            rtol=uvdata._freq_array.tols[0],
+        )
+    except ValueError:
+        freq_arr_match = False
+
+    if freq_arr_match is False:
```

Comment:
for similar reasons, it might be nice to also have an `override_freq_check` parameter

Reply:
We were unaware of a case when you would want to do such a thing -- bandpass calibration is usually quite frequency specific, whereas transferring cals between different times is something we've all seen before.

Reply:
The feeling on the telecon is that in very experimental modes like this the user can do the select on the objects before hand or even overwrite the frequency array on one of the objects if that's really what they want. The slow part of the check doesn't come up at all if the freq_array is matched between the UVCal and UVData objects.

Reply:
okay yup I'm fine w/ that too if the slow parts are only entered if the frequencies aren't matched